statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/state/MultiplexedState.java

```diff
@@ -65,6 +67,19 @@ public MultiplexedState(
     return new MultiplexedMapStateAccessor<>(sharedMapStateHandle, uniqueSubKey, valueSerializer);
   }
 
+  @Override
+  public <K, V> TableAccessor<K, V> createFlinkStateTableAccessor(
+      FunctionType functionType, PersistedTable<K, V> persistedTable) {
+    final MultiplexedStateKey uniqueSubKeyPrefix =
+        multiplexedSubstateKey(functionType, persistedTable.name());
+    final TypeSerializer<K> keySerializer =
+        types.registerType(persistedTable.keyType()).createSerializer(executionConfiguration);
+    final TypeSerializer<V> valueSerializer =
+        types.registerType(persistedTable.valueType()).createSerializer(executionConfiguration);
+    return new MultiplexedTableStateAccessor<>(
+        sharedMapStateHandle, uniqueSubKeyPrefix, keySerializer, valueSerializer);
```

Comment:
Some more food for thought:
If we do decide to separate the handles, we can slim down the `MultiplexedStateKey` type a bit, by having a separate `MultiplexedTableStateKey` that has a `ByteString userKey` field and a `MultiplexedStateKey prefix` field.

For example, I already have a minor concern with the way we use `MultiplexedStateKey`:
Does protobuf repeated fields require some extra metadata written? If yes, its a tad bit redundant size-wise in this case since we only ever have 1 user key added.

Reply:
Hi @tzulitai, 
I am hesitant to add another state handle without a clear benefit mainly because of the memory requirements associated with it. 
We Currently have:
* async operations state
* delayed operations
* persisted values
* there would be another ListState coming for multi-language
This already requires 512MB out of the box, so I'd be carful of adding another without a clear advantage. 
Ideally the core limitation at Flink would be addressed so we won't have to multiplex state at all.